### PR TITLE
Refine publication view

### LINKED ISSUE #3740: Iteration on publication gallery
Now that we have the first version of the publication gallery (which already is great!), we can do some iterations on it. What comes to my mind:

- [x] Add dataset size (in nm, µm or mm)
- [ ] Add Publication date
- [ ] Add Organization name
- [x] Full width list layout instead of 2-columns
- [x] Rename heading to "Featured publications"
- [x] How can I get to the frontpage when I'm logged in?
- [x] Hide Publication gallery when empty #3733 
- [x] Sort datasets in publication by sorting key

### PR DESCRIPTION
**Pull request by daniel-wer**
This PR refines the publication view as detailed in #3740 and discussed in the recent publication view redesign session. It also introduces a new feature flag `enableFrontpage` which will be enabled on demo and modifies what users see when visiting the main page `/`.
With `enableFrontpage = true`, `/` will always show the /spotlight page (publications view for now, but could contain other things like pricing etc later).
When disabled, `/` will show the dashboard when logged in (as before) and the login view when logged out (was spotlight before, but that didn't have much use as it is empty most of the time).
Users get to `/` when clicking on the webknossos logo or string in the top left corner.

Regardless, the publications view can always be visited by going to `/spotlight`.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Check out locally and insert publications and dataset details using pgadmin:
```sql
insert into webknossos.publications(_id, publicationDate, imageUrl, title, description) values('5c3c9ec895010095014759fd',NOW(),'https://webknossos.org/images/oxalis.svg','Dummy Title that is usually very long and contains highly scientific terms', 'This is a wonderful dummy publication, it has authors, it has a link, it has a doi number, those could go here');

update webknossos.datasets set _publication = '5c3c9ec895010095014759fd', details='{"species":"Mouse", "brainRegion":"lateral parietal association cortex P46", "acquisition":"Raw CLSM data"}' where _id = '<INSERT_DATASET_ID>' ;
```
- Repeat the update clause for as many dataset ids that you like. Take care when visiting the spotlight page while being logged out - only public datasets can be seen in that case.

- Check that all items in #3740 are done

### Issues:
- fixes #3740 

------
~~- [ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
~~- [ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
~~- [ ] Updated [documentation](../blob/master/docs) if applicable~~
~~- [ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

**Comment by normanrz** [comment]
I reworked the descriptions on demo.webknossos.org as per our discussion yesterday

**Comment by daniel-wer** [comment]
This should be ready for review now, but needs to be tested locally, see the PR description for how to insert publications and dataset details :)

This is what the view currently looks like:

publications

File: frontend/javascripts/dashboard/publication_card.js [hunk 1]
```diff
@@ -136,70 +153,72 @@ class PublicationCard extends React.PureComponent<Props, State> {
...
+                  className="dataset-thumbnail-image absolute segmentation"
+                  style={{
+                    backgroundImage: `url('${segmentationThumbnailURL}?w=${thumbnailDimension}&h=${thumbnailDimension}')`,
+                  }}
+                />
+              ) : null}
+              <ThumbnailOverlay details={details} />
+              {sortedDatasets.length > 1 && (
+                <div className="datasets-scrollbar-spacer">
+                  <div className="dataset-published-grid">
```
[the remark this answers is not in the document]
Comment (daniel-wer) [reply frontend/javascripts/dashboard/publication_card.js]: Yes, absolutely :)

**Comment by daniel-wer** [comment]
Thanks for the thorough feedback and for experimenting and tweaking yourself as well!

Regarding (1) and (2), I did clean up the "intertwinement" of the publication and dataset lists in this PR as I removed the publication view from the dashboard. The search bar was actually the one in the `spotlight_view` so it was very easy to adapt the max-width :)

Regarding the scroll bar, I'd say it doesn't need to be inverted, works pretty well as is:

scrollbar

I like the different header style, the "Featured Publications" headline feels kind of lost where it is now. What styles did you use for that?

**Pull request merged**